### PR TITLE
New obs message and lock time (try 2)

### PR DESCRIPTION
**Pull request by denniszollo**

### CODE REVIEW

File: src/base_obs.c [hunk 1]
```diff
@@ -417,6 +418,13 @@ void base_obs_setup()
     &deprecated_callback,
     &deprecated_node
   );
+  
+  static sbp_msg_callbacks_node_t deprecated_node_2;
+  sbp_register_cbk(
+    SBP_MSG_OBS_DEP_B,
```
Comment (mfine) [inline src/base_obs.c]: Maybe drop `SBP_MSG_OBS_DEP_A` - and just put out a deprecated warning for `SBP_MSG_OBS_DEP_B`, not both.

File: src/track/track_gps_l1ca.c [hunk 2]
```diff
@@ -140,7 +140,6 @@ static void tracker_gps_l1ca_init(const tracker_channel_info_t *channel_info,
 
   memset(data, 0, sizeof(gps_l1ca_tracker_data_t));
   tracker_ambiguity_unknown(channel_info->context);
-
```
Comment (jacobmcnamee) [inline src/track/track_gps_l1ca.c]: remove

File: src/solution.c [hunk 3]
```diff
@@ -762,7 +762,7 @@ void process_matched_obs(u8 n_sds, gps_time_t *t, sdiff_t *sds, u16 base_id)
   }
 }
 
-static THD_WORKING_AREA(wa_time_matched_obs_thread, 20000);
+static THD_WORKING_AREA(wa_time_matched_obs_thread, 18000);
```
Comment (jacobmcnamee) [inline src/solution.c]: Remove this change. The heap issue is likely not related to this PR, but we need to fix it separately and in a less hacky manner.

File: src/base_obs.c (hunk 1 above)
Comment (denniszollo) [inline src/base_obs.c]: disagree - we should send the warning no matter what.

Comment (mfine) [inline src/base_obs.c]: After a deprecation period, we should be comfortable removing warnings. We did for the old unpacked obs messages. If it's not clear, I'm saying warn on `DEP_B`, not also `DEP_A` (note this is also generally a poor way to do "version" checking between devices).

File: src/sbp_utils.c [hunk 4]
```diff
@@ -188,6 +188,30 @@ void sbp_make_heading(msg_baseline_heading_t *baseline_heading, const gps_time_t
...
+       for(u8 i = 0; i < 16; i++) {
+         if ((int)lock_time > (1<<(i+5))) {
+           continue;
+         }
+         else
+           return i;
+        }
+     }
+     return 15;
+}
```
Comment (mfine) [inline src/sbp_utils.c]: Style is off on these two (indentation, curly braces). Logic could also be simplified a little bit:

```c
u32 min_lock_time_decode(u8 lock_time_packed) {
  if (lock_time_packed == 0)
    return 0;

  return 1 << (lock_time_packed + 4);
}

u8 lock_time_encode(u32 lock_time) {
  for(u8 i = 0; i < 16; i++)
    if (lock_time <= (u32)(1 << (i+5)))
      return i;

  return 15;
}
```

Would be good to add some unit tests for these as well.

**Comment by denniszollo** [comment]
deprecated in favor of a soon to exist PR.

**Comment by denniszollo** [comment]
superceded by #706

**Pull request closed without merging**